holoviews/plotting/bokeh/util.py

```diff
@@ -193,6 +163,39 @@ def get_ids(obj):
     return list(itertools.chain(*ids))
 
 
+def replace_models(obj):
+    """
+    Processes references replacing Model and HasProps objects.
+    """
+    if isinstance(obj, Model):
+        return obj.ref
+    elif isinstance(obj, HasProps):
+        return obj.properties_with_values(include_defaults=False)
+    elif isinstance(obj, dict):
+        return {k: replace_models(v) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [replace_models(v) for v in obj]
+    else:
+        return obj
+
+
+def to_references(doc):
+    """
+    Convert the document to a dictionary of references.  Avoids
+    converting document to json and the performance penalty that
+    involves.
+    """
```

Comment:
Perhaps 'Avoids unnecessary JSON serialization/deserialization within Python and the corresponding performance penalty'. If you don't want to use that, just make it clear that this is all within Python.